monitor.cpp

```diff
@@ -1,38 +1,37 @@
 #include "./monitor.h"
-#include <assert.h>
 #include <thread>
 #include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include "./temperatureBodylimits.h"  // Include the header file for temperature body limits
+#include "./pulseratebodylimits.h"  // Include the header file for pulse rate body limits
+#include "./spo2bodylimits.h"  // Include the header file for spo2 body limits
+using std::cout;
+using std::flush;
+using std::this_thread::sleep_for;
+using std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+static void measureVitals() {
+    for (int i = 0; i < 1; i++) {
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+}
+
+static bool checkVital(const char* vitalName, float value, float min, float max) {
+    if (value < min || value > max) {
+        measureVitals();
+        cout << vitalName << " is out of range!\n";
+        return false;
     }
-    return 0;
-  }
-  return 1;
+    return true;
+}
+
+int vitalsOk(float temperature, float pulseRate, float spo2) {
+    bool temperatureOk = checkVital("Temperature", temperature, MIN_TEMPERATURE, MAX_TEMPERATURE);
+    bool pulseRateOk = checkVital("Pulse Rate", pulseRate, MIN_PULSE_RATE, MAX_PULSE_RATE);
+    bool spo2Ok = checkVital("Oxygen Saturation", spo2, MIN_SPO2, MAX_SPO2);
+
```

Comment:
Can also see `std::all_of`, to restrict complexity (and a long conjunction) in case parameters increase